sources/api/netdog/src/net_config/static_address.rs

```diff
@@ -48,6 +50,19 @@ impl TryFrom<String> for RouteTo {
     }
 }
 
+impl Validate for StaticConfigV1 {
+    fn validate(&self) -> ValidateResult<()> {
+        ensure!(
+            self.addresses.iter().all(|a| matches!(a, IpNet::V4(_)))
+                || self.addresses.iter().all(|a| matches!(a, IpNet::V6(_))),
+            InvalidNetConfigSnafu {
+                reason: "static configuration must only contain all IPv4 or all IPv6 addresses"
```

Comment:
I think this answers a question I asked elsewhere.

In general I'm sad to see all of this validation code. And I'm sure I was asked for help with this and probably didn't come up with anything, but there's always a way to avoid "validate after construction", especially with Rust. Just something to keep in mind in future designs. 

(Also, I realize that making it compatible with the serialization shape is challenging)

Instead of `Vec<V4orV6Enum`...


```
enum IpV4OrIpV6List {
    IpV4(Vec<IpV4Struct>),
    IpV6(Vec<IpV6Struct>),
}
```

Which is one example of eliminating a `validate()` call.